stdlib/internals.ncl

```diff
@@ -45,29 +45,49 @@
   "$enum_fail" = fun l =>
       %blame% (%tag% "tag not included in the enum type" l),
 
-  "$record" = fun cont l t =>
-      if %typeof% t == `Record then
-          %assume% (cont {}) l t
+  "$record" = fun field_contracts tail_contract l t =>
```

Comment:
I'm not thrilled with how big `$record` is now, but I also wasn't sure how best to split it up (or whether it even really warrants it). Let me know if you have any ideas!